sdk/tables/aztables/tableClient.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package aztables
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	//"net/url"
+	//"time"
+)
+
+// A TableClient represents a URL to an Azure Storage blob; the blob may be a block blob, append blob, or page blob.
+type TableClient struct {
+	client *tableClient
+	cred   SharedKeyCredential
+}
+
+// NewTableClient creates a TableClient object using the specified URL and request policy pipeline.
+func NewTableClient(serviceURL string, cred azcore.Credential, options *ClientOptions) (TableClient, error) {
```

Comment:
I'm wondering if ClientOptions should be called TableClientOptions? Thoughts? Would there ever be any other client in this module? Is this what azblob did?

Reply:
I just checked azblob, I see they left it as ClientOptions.

Reply:
It was my intuition to name it `TableClientOptions` as well. I did copy this naming from azblob, but I also prefer the more specific name if `ClientOptions` is not a convention.

Reply:
I'm not sure, I can see why it's just client options...since it'll probably only ever have options related to modifying the policies. So I'm unsure. I agree that my first instinct was to also go with `TableClientOptions`. I think this might be a better question for Jeff when he sees this.

Reply:
Does `azblob` share the `ClientOptions` across all its clients?  If so, that's probably why they went with a generic name.

Reply:
Yes I think so. I think the naming for the options will depend on how many other clients there will be and what options they'll have and if in the future we think that the options could vary between clients to a degree where we should have individually named options for client.

Reply:
changing to to TableClientOptions for now - we can always revisit later if needed.